### PR TITLE
Get extensions from .cabal file.

### PR DESCRIPTION
**Pull request by AshleyYakeley**

### CODE REVIEW

**Comment by AshleyYakeley** [comment]
Fix for #461.

**Comment by AshleyYakeley** [comment]
Please hold off on merging this, there are still problems with it.

File: src/HIndent/CabalFile.hs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            ParseFailed _ -> return []
+            ParseOk _ gpd -> do
+              return $ allBuildInfo' $ flattenPackageDescription gpd
+      let bis = filter (matchBuildInfo relpath) $ mconcat biss
+      return $
+        fmap fst $
+        getBest $ do
+          bi <- bis
+          dir <- hsSourceDirs bi
+          -- the best one is the one with the longest hsSourceDirs
```
Comment (AshleyYakeley) [inline src/HIndent/CabalFile.hs]: This fails to distinguish stanzas with the same hs-source-dirs.

Comment (AshleyYakeley) [reply src/HIndent/CabalFile.hs]: Fixed now.

**Comment by chrisdone** [comment]
Cool!

**Comment by chrisdone** [comment]
Seems to work fine. Thanks!

**Pull request merged**